addons/project_timesheet_holidays/models/project_task.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+
+class Task(models.Model):
+    _inherit = 'project.task'
+
+    timeoff_type_ids = fields.One2many('hr.leave.type', 'timesheet_task_id', string="Task Time off Types")
+    is_timeoff_task = fields.Boolean(string="Is Time off Task", compute="_compute_is_timeoff_task",
+        search="_search_is_timeoff_task")
+
+    @api.depends('timeoff_type_ids', 'company_id.leave_timesheet_task_id')
+    def _compute_is_timeoff_task(self):
+        timeoff_tasks = self.filtered(lambda task: task.timeoff_type_ids or task.company_id.leave_timesheet_task_id == task)
```

Comment:
It could be better to count the `timeoff_type_ids` instead of fetching all to know if a task has at least one record.
So I suggest to create new field in which we will have the number of timeoff types.

Here is the changes with the suggestion below. 🙂 
```diff
diff --git a/addons/project_timesheet_holidays/models/project_task.py b/addons/project_timesheet_holidays/models/project_task.py
index 9a6cc60af37..5548748f3ea 100644
--- a/addons/project_timesheet_holidays/models/project_task.py
+++ b/addons/project_timesheet_holidays/models/project_task.py
@@ -7,15 +7,24 @@ class Task(models.Model):
     _inherit = 'project.task'
 
     timeoff_type_ids = fields.One2many('hr.leave.type', 'timesheet_task_id', string="Task Time off Types")
+    leave_types_count = fields.Integer(compute='_compute_leave_types_count')
     is_timeoff_task = fields.Boolean(string="Is Time off Task", compute="_compute_is_timeoff_task",
         search="_search_is_timeoff_task")
 
-    @api.depends('timeoff_type_ids', 'company_id.leave_timesheet_task_id')
-    def _compute_is_timeoff_task(self):
-        timeoff_tasks = self.filtered(lambda task: task.timeoff_type_ids or task.company_id.leave_timesheet_task_id == task)
-        timeoff_tasks.update({'is_timeoff_task': True})
+    def _compute_leave_types_count(self):
+        time_off_type_read_group = self.env['hr.leave.type']._read_group(
+            [('timesheet_task_id', 'in', self.ids)],
+            ['task_id'],
+            ['task_id'],
+        )
+        time_off_type_count_per_task = {res['task_id'][0]: res['task_id_count'] for res in time_off_type_read_group}
+        for task in self:
+            task.leave_types_count = time_off_type_count_per_task.get(task.id, 0)
 
-        (self - timeoff_tasks).update({'is_timeoff_task': False})
+    def _compute_is_timeoff_task(self):
+        timeoff_tasks = self.filtered(lambda task: task.leave_types_count or task.company_id.leave_timesheet_task_id == task)
+        timeoff_tasks.is_timeoff_task = True
+        (self - timeoff_tasks).is_timeoff_task = False
 
     def _search_is_timeoff_task(self, operator, value):
         if operator not in ['=', '!='] or not isinstance(value, bool):
```